src/legacy/ui/public/chrome/api/sub_url_hooks.js

```diff
@@ -19,18 +19,23 @@
 
 import url from 'url';
 
-import {
-  unhashUrl,
-} from '../../state_management/state_hashing';
+import { unhashUrl } from '../../../../../plugins/kibana_utils/public';
+import { toastNotifications } from '../../notify/toasts';
 
 export function registerSubUrlHooks(angularModule, internals) {
   angularModule.run(($rootScope, Private, $location) => {
     const subUrlRouteFilter = Private(SubUrlRouteFilterProvider);
 
     function updateSubUrls() {
       const urlWithHashes = window.location.href;
-      const urlWithStates = unhashUrl(urlWithHashes);
-      internals.trackPossibleSubUrl(urlWithStates);
+      let urlWithStates;
+      try {
+        urlWithStates = unhashUrl(urlWithHashes);
+      } catch (e) {
+        toastNotifications.addDanger(e.message);
```

Comment:
fixes small regression introduced in #51835,
When sharing hashed url directly it should show error toast instead of full page fatal error